src/fuse_pointwise.cpp

```diff
@@ -189,6 +192,53 @@ static bool find_pointwise_modules(module& m)
     }
     return changed;
 }
+namespace {
+struct find_pointwise_reshape_pointwise
+{
+    auto matcher() const
+    {
+        auto reshape =
+            match::name("reshape", "squeeze", "unsqueeze", "flatten")(match::used_once());
+        auto skip_contiguous =
+            match::arg(0)(match::skip(match::name("contiguous")(match::used_once())));
+        auto pointwise         = match::name("pointwise")(match::used_once());
+        auto reshape_pointwise = reshape(skip_contiguous(pointwise.bind("x"))).bind("reshape");
+        return match::name("pointwise")(match::any_of[match::inputs()](reshape_pointwise));
+    }
+
+    void apply(module& m, const match::matcher_result& r) const
+    {
+        auto ins         = r.result;
+        auto x_ins       = r.instructions["x"];
+        auto reshape_ins = r.instructions["reshape"];
+
+        auto cd = common_dims::compute(ins->get_shape().lens(), x_ins->get_shape().lens());
+        if(cd.dims.empty())
+            return;
+
+        auto reshape_input = [&](const auto& ins_to_insert) {
+            return [&](auto input) {
+                auto c = m.insert_instruction(ins_to_insert, make_op("contiguous"), input);
```

Comment:
Would this be better performance-wise if we have to add `contiguous` for all the inputs? 

`eliminate_contigous` would eliminate some of them but if the inputs are non-standard shaped for the first pointwise module, it could add additional overhead. 

Perhaps we should just limit this fusion to the standard-shaped inputs.

Reply:
> Would this be better performance-wise if we have to add contiguous for all the inputs?

Thats why I use the common_dims to prevent the need to add contiguous. 

> Perhaps we should just limit this fusion to the standard-shaped inputs.

We dont know if the inputs will be standard shape until after `eliminate_contiguous`.